### PR TITLE
refactor: missing localization in components

### PR DESCRIPTION
**Pull request by Highjhacker**
## Summary

https://app.clickup.com/t/we6z11

## Checklist

-   [x] I checked my UI changes against the design and there are no notable differences
-   [x] I checked my UI changes for any responsiveness issues
-   [x] I checked my (code) changes for obvious issues, debug statements and commented code
-   [ ] I provided a screenshot of my changes to the component _(if applicable)_
-   [ ] I regenerated the `icons.html` file and checked if my newly added icon is shown correctly _(if necessary)_
-   [ ] I added an explanation on how to use the component to the readme _(if necessary)_
-   [ ] Documentation _(if necessary)_
-   [ ] Tests _(if necessary)_
-   [x] Ready to be merged

### CODE REVIEW

File: resources/lang/en/numeric.php [hunk 1]
```diff
@@ -0,0 +1,7 @@
+<?php
+
+declare(strict_types=1);
+
+return [
+    'mb' => 'MB',
```
Comment (ItsANameToo) [inline resources/lang/en/numeric.php:6]: move it to generic, no use having a `numeric` lang file that only contains `mb` (filename made me think of other values than "MB" in here too)

File: resources/views/inputs/upload.blade.php [hunk 2]
```diff
@@ -34,12 +34,12 @@ class="block absolute top-0 opacity-0 cursor-pointer"
...
                                 @if ($click ?? false) @click="{{ $click }}" @endif
                             >
-                                Upload a file
+                                @lang('ui::actions.upload_file')
                             </button>
                         </p>
 
                         <p class="mt-1 text-xs text-theme-secondary-500">
-                            {{ strtoupper(implode(', ', $extensions ?? [])) }} up to {{ $size ?? '10' }}MB
+                            {{ strtoupper(implode(', ', $extensions ?? [])) }} @lang('ui::generic.up_to') {{ $size ?? '10' }}@lang('ui::numeric.mb')
```
Comment (ItsANameToo) [inline resources/views/inputs/upload.blade.php:42]: wouldn't it be better to make this a single text with parameters? Using a lang file while hardcoding values like that defeats the purpose of having them as they can change between languages (e.g. $1.00 vs 1,00$ as example)

File: resources/views/pagination-url.blade.php [hunk 3]
```diff
@@ -33,7 +33,7 @@ class="py-2 px-3 w-full bg-transparent dark:text-theme-secondary-200"
             class="button-pagination-page-indicator button-pagination-page-indicator--search"
             :class="{ 'opacity-0': search }"
             x-on:click="toggleSearch"
-        ><span>Page {{ $paginator->currentPage() }} of {{ $paginator->lastPage() }}</span></button>
+        ><span>@lang('ui::generic.page') {{ $paginator->currentPage() }} @lang('ui::generic.of') {{ $paginator->lastPage() }}</span></button>
```
Comment (ItsANameToo) [inline resources/views/pagination-url.blade.php:36]: make it a single lang entry with parameters

File: resources/views/pagination-url.blade.php [hunk 4]
```diff
@@ -128,22 +128,22 @@ class="@if ($paginator->currentPage() === $page) button-pagination-page-indicato
                     type="button"
                     class="button-pagination-page-indicator button-pagination-page-indicator--search"
                     :class="{ 'opacity-0': search }"
-                ><span>Page {{ $paginator->currentPage() }} of {{ $paginator->lastPage() }}</span></button>
+                ><span>@lang('ui::generic.page') {{ $paginator->currentPage() }} @lang('ui::generic.of') {{ $paginator->lastPage() }}</span></button>
```
Comment (ItsANameToo) [inline resources/views/pagination-url.blade.php:131]: single entry

File: resources/views/pagination.blade.php [hunk 5]
```diff
@@ -33,7 +33,7 @@ class="py-2 px-3 w-full bg-transparent dark:text-theme-secondary-200"
             class="button-pagination-page-indicator button-pagination-page-indicator--search"
             :class="{ 'opacity-0': search }"
             x-on:click="toggleSearch"
-        ><span>Page {{ $paginator->currentPage() }} of {{ $paginator->lastPage() }}</span></button>
+        ><span>@lang('ui::generic.page') {{ $paginator->currentPage() }} @lang('ui::generic.of') {{ $paginator->lastPage() }}</span></button>
```
Comment (ItsANameToo) [inline resources/views/pagination.blade.php:36]: single entry

File: resources/views/pagination.blade.php [hunk 6]
```diff
@@ -119,7 +119,7 @@ class="@if ($paginator->currentPage() === $page) button-pagination-page-indicato
                     type="button"
                     class="button-pagination-page-indicator button-pagination-page-indicator--search"
                     :class="{ 'opacity-0': search }"
-                ><span>Page {{ $paginator->currentPage() }} of {{ $paginator->lastPage() }}</span></button>
+                ><span>@lang('ui::generic.page') {{ $paginator->currentPage() }} @lang('ui::generic.of') {{ $paginator->lastPage() }}</span></button>
```
Comment (ItsANameToo) [inline resources/views/pagination.blade.php:122]: single entry

File: resources/lang/en/numeric.php (hunk 1 above)
Comment (Highjhacker) [reply resources/lang/en/numeric.php:6]: Yeah, fair point, shouldn't be more than that, was just to cater about the weird french translations (MB => MO) but yeah, gonna move it to generic.

File: resources/views/inputs/upload.blade.php [hunk 7]
```diff
@@ -34,12 +34,13 @@ class="block absolute top-0 opacity-0 cursor-pointer"
...
                             >
-                                Upload a file
+                                @lang('ui::actions.upload_file')
                             </button>
                         </p>
 
                         <p class="mt-1 text-xs text-theme-secondary-500">
-                            {{ strtoupper(implode(', ', $extensions ?? [])) }} up to {{ $size ?? '10' }}MB
+                            {{ strtoupper(implode(', ', $extensions ?? [])) }} @lang('ui::generic.upload.up_to', ['size' => $size ?? 10, 'filesize' => trans('ui::generic.filesize.mb')])
+                            {{ strtoupper(implode(', ', $extensions ?? [])) }} @lang('ui::generic.up_to') {{ $size ?? '10' }}@lang('ui::generic.filesize.mb')
```
Comment (alexbarnsley) [inline resources/views/inputs/upload.blade.php:43]: is this supposed to be here? :thinking:

File: resources/views/pagination-url.blade.php [hunk 8]
```diff
@@ -33,7 +33,7 @@ class="py-2 px-3 w-full bg-transparent dark:text-theme-secondary-200"
             class="button-pagination-page-indicator button-pagination-page-indicator--search"
             :class="{ 'opacity-0': search }"
             x-on:click="toggleSearch"
-        ><span>Page {{ $paginator->currentPage() }} of {{ $paginator->lastPage() }}</span></button>
+        ><span>@lang('ui::generic.pagination.current_to', ['currentPage' => $paginator->currentPage(), 'lastPage' => $paginator->lastPage()])</span></button>
```
Comment (alexbarnsley) [inline resources/views/pagination-url.blade.php:36]: Not sure if the elements need to be next to each other or not so i left them, but the line was too long so split up the array

```suggestion
        ><span>@lang('ui::generic.pagination.current_to', [
        	'currentPage' => $paginator->currentPage(),
        	'lastPage' => $paginator->lastPage(),
        ])</span></button>
```

File: resources/views/pagination-url.blade.php [hunk 9]
```diff
@@ -128,22 +128,22 @@ class="@if ($paginator->currentPage() === $page) button-pagination-page-indicato
                     type="button"
                     class="button-pagination-page-indicator button-pagination-page-indicator--search"
                     :class="{ 'opacity-0': search }"
-                ><span>Page {{ $paginator->currentPage() }} of {{ $paginator->lastPage() }}</span></button>
+                ><span>@lang('ui::generic.pagination.current_to', ['currentPage' => $paginator->currentPage(), 'lastPage' => $paginator->lastPage()])</span></button>
```
Comment (alexbarnsley) [inline resources/views/pagination-url.blade.php:131]:
```suggestion
                ><span>@lang('ui::generic.pagination.current_to', [
                	'currentPage' => $paginator->currentPage(),
                	'lastPage' => $paginator->lastPage(),
                ])</span></button>
```

File: resources/views/pagination.blade.php [hunk 10]
```diff
@@ -33,7 +33,7 @@ class="py-2 px-3 w-full bg-transparent dark:text-theme-secondary-200"
             class="button-pagination-page-indicator button-pagination-page-indicator--search"
             :class="{ 'opacity-0': search }"
             x-on:click="toggleSearch"
-        ><span>Page {{ $paginator->currentPage() }} of {{ $paginator->lastPage() }}</span></button>
+        ><span>@lang('ui::generic.pagination.current_to', ['currentPage' => $paginator->currentPage(), 'lastPage' => $paginator->lastPage()])</span></button>
```
Comment (alexbarnsley) [inline resources/views/pagination.blade.php:36]:
```suggestion
        ><span>@lang('ui::generic.pagination.current_to', [
        	'currentPage' => $paginator->currentPage(),
        	'lastPage' => $paginator->lastPage(),
        ])</span></button>
```

File: resources/views/pagination.blade.php [hunk 11]
```diff
@@ -119,7 +119,7 @@ class="@if ($paginator->currentPage() === $page) button-pagination-page-indicato
                     type="button"
                     class="button-pagination-page-indicator button-pagination-page-indicator--search"
                     :class="{ 'opacity-0': search }"
-                ><span>Page {{ $paginator->currentPage() }} of {{ $paginator->lastPage() }}</span></button>
+                ><span>@lang('ui::generic.pagination.current_to', ['currentPage' => $paginator->currentPage(), 'lastPage' => $paginator->lastPage()])</span></button>
```
Comment (alexbarnsley) [inline resources/views/pagination.blade.php:122]:
```suggestion
                ><span>@lang('ui::generic.pagination.current_to', [
                	'currentPage' => $paginator->currentPage(),
                	'lastPage' => $paginator->lastPage(),
                ])</span></button>
```

**Pull request merged**